code/datums/status_effects/debuffs.dm

```diff
@@ -951,6 +951,12 @@
 
 /datum/status_effect/heretic_mark/rust
 	effect_icon_state = "emark3"
+	var/static/list/item_blacklist = list()
+
+/datum/status_effect/heretic_mark/rust/on_creation(mob/living/new_owner, ...)
+	. = ..()
+	if(!item_blacklist[owner])
+		item_blacklist[owner] = typecacheof(list(/obj/item/grenade))
```

Comment:
Why is the typecache per-owner when its exactly the same for every mob?
Also if there is only 1 thing in the typecache, it would be better to just save the memory and initialisation cost and check istype(thing, /obj/item/grenade) instead of doing a typecache

Reply:
So I would actually just return to the original code for this, since it is unlikely to be expanded and can be altered to a typecache if it is expanded.

Reply:
My bad, I recommended a typecache under the impression there were multiple things required to be blacklisted

I also don't really know what the owner thing is about :(

Reply:
Typecaches are good for runtime performance, but bad for initialisation performance. They are good in cases where you are going to be performing the lookups extremely frequently but its usually not worth the cost if its something that is triggered under rarer circumstances.

Reply:
alright, reverting typecache changes